### ISSUE TITLE
URL de recherche directe

### ISSUE THREAD

**Issue by XioNoX** [laem/futureco#317]
Serait-ce possible que https://cartes.app/?allez=Rennes  ouvre cartes.app directement sur Rennes ? (ou similaire)

Actuellement pour trouver un lieu il faut ouvrir cartes.app, puis taper un lieu, par exemple "Rennes", puis attendre quelques secondes que la liste de suggestions charge, puis cliquer sur le 1er nom qui sort.

Pour comparer, avec Google Maps c'est possible de naviguer directement sur https://www.google.fr/maps/?q=Rennes
Ce qui permet d'avoir des "smart keywords" https://support.mozilla.org/fr/kb/comment-rechercher-site-barre-adresse

Merci !

**Comment by laem** [comment]
Ah, dslé j'avais pas vu ton message. C'est déjà possible ! La syntaxe est la suivante : ?q=Rennes. 

Par contre il n'est pas (encore) intelligent, il ne va pas centrer automatiquement cartes.app/?q=rennes

**Comment by laem** [comment]
Je vois que Google fait des choix partials : https://fr.wikipedia.org/wiki/Camaret ou encore https://fr.wikipedia.org/wiki/Montreuil. Il dirige directement sur une de ces villes.

**Comment by laem** [comment]
Plusieurs paramères pour l'algo : 

- si 90% des points trouvés sont dans une zone, zoomer sur la zone
- si le score du premier élément de la recherche Photon est élevé (comment qualifier ça d'élevé ?), choix automatique du premier avec petit encart [étendre la recherche aux autre "Rennes"]

**Comment by bohwaz** [comment]
Effectivement ça manque pour permettre d'utiliser Cartes.app depuis un site web, en lien pour renvoyer. Aujourd'hui le "?q=" ne fait qu'afficher une liste de lieux trouvés, mais il faudrait centrer sur le premier lieu de la liste.

Openstreetmap.org le fait très bien.

**Comment by laem** [comment]
> Openstreetmap.org le fait très bien.

Non, OSM le fait trop souvent par erreur. CF par exemple la recherche "loire" au zoom sur la France. Choisir pour l'utilisateur de zoomer sur le département est une erreur. 

La bonne réponse me semble celle de Google maps, qui zoom sur la Loire, le fleuve, aux environs de la Loire à vélo avec Nantes et Tours pas loin.

Cf ma réponse précédente, ici même... https://github.com/cartesapp/cartes/issues/317#issuecomment-2174242433
